doc/kubernetes/modules/ROOT/pages/running/timeout_tunning.adoc

```diff
@@ -0,0 +1,103 @@
+= Understanding failure detection in Infinispan and JGroups
+:description: This guide describes how JGroups detects failed instances and its configuration.
+
+{description}
+
+== Audience
+
+This guide describe the failure detection mechanism that JGroups enforces to detect crashed pods inside the Infinispan cluster.
+It explains the configuration attributes available for customization and its trade-off to achieve the desired detection rate.
+
+== Broken TCP connections
+
+A broken TCP connection have a performance impact in the cluster, and it is necessary to detect it so the connection can be closed and new one may be estabilished.
+Note that TCP is a blocking/synchronous protocol, and if the connection is not working properly, threads will be blocked in the `Socket::write` operation.
+The kernel parameter `tcp_retries2` configures the amount failed retries until the Kernel force closes the connection.
+Quoting the https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt[Kernel Documentation]:
+
+----
+This value influences the timeout of an alive TCP connection, when RTO retransmissions remain unacknowledged.
+The default value of 15 yields a hypothetical timeout of 924.6 seconds and is a lower bound for the effective timeout.
+----
+
+During this period, the cluster performance is degraded.
+Changing Kernel parameters in image or container is not desired, but JGroups has attributes that can be configured to detect and force-close these broken connections.
+
+For cross-site network communication, which use the http://jgroups.org/manual5/index.html#TUNNEL[TUNNEL] protocol in Kubernetes, Infinispan pod traffic is forwarded through JGroups Gossip Router pods.
+The timeouts are tuning in the `Infinispan CR` YAML file as shown below:
+
+.infinispan.yaml
+[source,yaml]
+----
+  service:
+    type: DataGrid
+    sites:
+      local:
+        discovery:
+          heartbeats:
+            interval: 10000 # <1>
+            timeout: 30000 # <2>
+----
+
+<1> Sends a heartbeat to the GossipRouter every `interval` milliseconds.
+<2> Max time in milliseconds with no received message or heartbeat after which the connection to a GossipRouter is closed.
+
+NOTE: The values above are the defaults configured by the Infinispan Operator.
+
+Reducing the interval and timeout will increase how fast JGroups detects a broken connection.
+However, it also increases the network usage due to the increase frequency of heartbeats.
+Note that all Infinispan pods in both cluster will send heartbeats to all the configured Gossip Router pods.
+
+TIP: The defaults should be work for 99% of the uses cases.
+
+For intra cluster communication in Infinispan or Keycloak cluster, broken TCP connection are detected and closed by the failure detection.
+See section <<fd>> for more information.
+
+[[fd]]
+== Failure Detection
```

Comment:
@belaban when you get a chance, can you take a look at this section

Reply:
* FD_ALL3: perhaps mention that doesn't send out heartbeats when regular messages have been sent

Reply:
We use mostly distributed caches (use unicast only) and heartbeats are suppressed only with multicast:

https://github.com/belaban/JGroups/blob/901197b381b6228eb3d2fde862972839812dfce6/src/org/jgroups/protocols/FailureDetection.java#L338

Is it worth mentioning?

Reply:
'Multicast' in this respect means that we're sending a heartbeat to *everyone*, so in TCP this is an N-1 operation